Library/Homebrew/formula.rb

```diff
@@ -222,10 +222,28 @@ def active_spec=(spec_sym)
     spec = send(spec_sym)
     raise FormulaSpecificationError, "#{spec_sym} spec is not available for #{full_name}" unless spec
 
+    old_spec_sym = @active_spec_sym
     @active_spec = spec
     @active_spec_sym = spec_sym
     validate_attributes!
     @build = active_spec.build
+
+    return if spec_sym == old_spec_sym
+
+    Dependency.clear_cache
+    Requirement.clear_cache
+  end
+
+  # @private
+  def build=(build_options)
+    old_options = @build
+    @build = build_options
+
+    return if old_options&.used_options == build_options&.used_options &&
+              old_options&.unused_options == build_options&.unused_options
```

Comment:
nil to nil comparisons here are fine I assume?

Reply:
I'll relook into this. I'm not really convinced this should ever be nil - it just was in some test.

Reply:
https://github.com/Homebrew/brew/blob/e143dfdad6bbd4cd178906eccac5de095c80dc2f/Library/Homebrew/test/formula_spec.rb#L168-L172

Not really convinced `nil` is something we should support. I don't see any `build` nil checks anywhere.